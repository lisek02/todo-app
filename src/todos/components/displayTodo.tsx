import * as React from 'react';
import { TodoModel } from '../models/todo.model';
import { RemoveItemPayload, ToggleEditPayload, ToggleCompletedPayload } from '../store/todoListActions';
import Checkbox from '@material-ui/core/Checkbox';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import CreateIcon from '@material-ui/icons/Create';

interface Props {
  todo: TodoModel;
  removeItem: (payload: RemoveItemPayload) => void;
  toggleEdit: (payload: ToggleEditPayload) => void;
  toggleCompleted: (payload: ToggleCompletedPayload) => void;
}

export const DisplayTodoComponent = (props: Props) => {
  const { todo, removeItem, toggleEdit, toggleCompleted } = props;

  return (
    <div>
      <Checkbox checked={todo.completed} onClick={() => toggleCompleted({ id: todo.id })} />
      <span>{ todo.title }</span>
      <ListItemSecondaryAction>
        <IconButton aria-label="Delete" onClick={() => removeItem({ id: todo.id })}>
          <DeleteIcon />
        </IconButton>
        <IconButton aria-label="Edit" onClick={() => toggleEdit({ id: todo.id })}>
          <CreateIcon />
        </IconButton>
      </ListItemSecondaryAction>
    </div>
  )
}
