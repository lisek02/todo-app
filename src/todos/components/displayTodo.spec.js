import * as React from 'react';
import { DisplayTodoComponent } from './displayTodo';
import { createShallow } from '@material-ui/core/test-utils';

const mockedTodo = { title: 'Title', id: 1 };
const mockedToggleCompleted = jest.fn();
let shallow;

describe('DisplayTodoComponent', () => {
  beforeAll(() => {
    shallow = createShallow();
  });

  it('should render correctly with todo as props', () => {
    const component = shallow(<DisplayTodoComponent todo={mockedTodo}/ >);
    
    expect(component).toMatchSnapshot();
  });

  it('should call toggleCompleted with { id: 1 } after clicking Checkbox', () => {
    const component = shallow(<DisplayTodoComponent todo={mockedTodo} toggleCompleted={mockedToggleCompleted} />);

    component
      .childAt(0)
      .simulate('click');

    expect(mockedToggleCompleted).toHaveBeenCalledWith({ id: 1 });
  });
});