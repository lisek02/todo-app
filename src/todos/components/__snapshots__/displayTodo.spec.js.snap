// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayTodoComponent should render correctly with todo as props 1`] = `
<div>
  <WithStyles(Checkbox)
    onClick={[Function]}
  />
  <span>
    Title
  </span>
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      aria-label="Delete"
      onClick={[Function]}
    >
      <pure(DeleteIcon) />
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      aria-label="Edit"
      onClick={[Function]}
    >
      <pure(CreateIcon) />
    </WithStyles(IconButton)>
  </WithStyles(ListItemSecondaryAction)>
</div>
`;

exports[`MyComponent should call toggleCompleted after clicking Checkbox 1`] = `
<div>
  <WithStyles(Checkbox)
    onClick={[Function]}
  />
  <span>
    Title
  </span>
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      aria-label="Delete"
      onClick={[Function]}
    >
      <pure(DeleteIcon) />
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      aria-label="Edit"
      onClick={[Function]}
    >
      <pure(CreateIcon) />
    </WithStyles(IconButton)>
  </WithStyles(ListItemSecondaryAction)>
</div>
`;

exports[`MyComponent should render correctly with no props 1`] = `
<div>
  <WithStyles(Checkbox)
    onClick={[Function]}
  />
  <span>
    Title
  </span>
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      aria-label="Delete"
      onClick={[Function]}
    >
      <pure(DeleteIcon) />
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      aria-label="Edit"
      onClick={[Function]}
    >
      <pure(CreateIcon) />
    </WithStyles(IconButton)>
  </WithStyles(ListItemSecondaryAction)>
</div>
`;

exports[`MyComponent should render correctly with todo as props 1`] = `
<div>
  <WithStyles(Checkbox)
    onClick={[Function]}
  />
  <span>
    Title
  </span>
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      aria-label="Delete"
      onClick={[Function]}
    >
      <pure(DeleteIcon) />
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      aria-label="Edit"
      onClick={[Function]}
    >
      <pure(CreateIcon) />
    </WithStyles(IconButton)>
  </WithStyles(ListItemSecondaryAction)>
</div>
`;
